### PR TITLE
feat: refactor config-bob build function, and add onepassword connect

### PR DESCRIPTION
**Pull request by smartinov**
- implement 1Password Connect
- implement 1Password CLI (for config-bob)
- cobra CLI

### CODE REVIEW

File: pkg/build/build.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	templateFunctions["secret"] = config.SecretManager.GetSecret
+
+	l.Info(line)
+	fmt.Println("building")
+	fmt.Println("data files     :", strings.Join(config.ValueFiles, ", "))
+	fmt.Println("source folders :", strings.Join(config.TemplatePaths, ", "))
+	fmt.Println(line)
+	data, err := readData(config.ValueFiles)
+	if err != nil {
+		return Result{}, errors.New("could not read data from: " + strings.Join(config.ValueFiles, ", ") + " :: " + err.Error())
```
Comment (pcicic) [inline pkg/build/build.go:47]: Wrapf!

File: pkg/build/build.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+	fmt.Println(line)
+	fmt.Println("building folder structure:")
+	fmt.Println(line)
+	err := os.MkdirAll(outputPath, 0744)
+	if err != nil {
+		return errors.New("could not create target folder")
+	}
+	i := 0
+	for _, folder := range result.Folders {
+		i++
```
Comment (pcicic) [inline pkg/build/build.go:84]: ?

File: pkg/build/build.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+			err = e
+		}
+		return !tartgetInfo.IsDir()
+	})
+	return
+}
+
+func getFolders(root string, ignore []string) (folders []string, err error) {
+	folders, err = filterFiles(root, ignore, func(path string, fileInfo os.FileInfo) bool {
+		targetInfo, e := resolve(fileInfo, path)
```
Comment (pcicic) [inline pkg/build/build.go:228]: err

File: pkg/build/build.go [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+		if trimmedPath == ignored {
+			return true
+		}
+	}
+	return false
+}
+
+func getFiles(root string, ignore []string) (files []string, err error) {
+	files, err = filterFiles(root, ignore, func(path string, fileInfo os.FileInfo) bool {
+		tartgetInfo, e := resolve(fileInfo, path)
```
Comment (pcicic) [inline pkg/build/build.go:217]: err

**Review by pcicic — COMMENTED** [review]
please update those few inherited inconsistencies

File: pkg/build/build.go (hunk 1 above)
Comment (SchumacherFM) [reply pkg/build/build.go:47]: `Wrapf` only makes sense when you need to inspect the error in the callee.

File: pkg/build/build.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+		fmt.Println(res.Info.Mode().Perm(), i, file)
+		err := os.WriteFile(file, res.Data, res.Info.Mode().Perm())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func readData(files []string) (interface{}, error) {
```
Comment (SchumacherFM) [inline pkg/build/build.go]: should be `(map[string]interface{},error)`

File: pkg/build/build.go [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+	if len(files) == 0 {
+		return nil, nil
+	}
+	data := make(map[string]interface{})
+
+	for _, file := range files {
+		fileData := make(map[string]interface{})
```
Comment (SchumacherFM) [inline pkg/build/build.go]: move this to line 131

File: pkg/build/build.go [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+	}
+	return p, nil
+}
+
+func getIgnore(root string) (ignore []string) {
+	ignore = []string{".bobignore", ".bobcopy"}
+	return append(ignore, getStuff(root, ".bobignore")...)
+}
+
+func getStuff(root, name string) []string {
```
Comment (SchumacherFM) [inline pkg/build/build.go:188]: what is stuff? please use correct name

File: pkg/build/build.go [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+	for _, ignored := range ignore {
+		if trimmedPath == ignored {
+			return true
+		}
+	}
+	return false
+}
+
+func getFiles(root string, ignore []string) (files []string, err error) {
+	files, err = filterFiles(root, ignore, func(path string, fileInfo os.FileInfo) bool {
```
Comment (SchumacherFM) [inline pkg/build/build.go:216]: `return filterFiles(...)`

File: pkg/build/build.go [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+	trimmedPath := strings.TrimPrefix(p, prefix)
+	for _, ignored := range ignore {
+		if trimmedPath == ignored {
+			return true
+		}
+	}
+	return false
+}
+
+func getFiles(root string, ignore []string) (files []string, err error) {
```
Comment (SchumacherFM) [inline pkg/build/build.go:215]: it is just `([]string,error)`

File: pkg/build/build.go [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+		tartgetInfo, e := resolve(fileInfo, path)
+		if e != nil {
+			err = e
+		}
+		return !tartgetInfo.IsDir()
+	})
+	return
+}
+
+func getFolders(root string, ignore []string) (folders []string, err error) {
```
Comment (SchumacherFM) [inline pkg/build/build.go:226]: same as `getFiles`

File: pkg/build/build.go [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+	}, nil
+
+}
+
+func process(templName, templ string, data interface{}, templateFuncMap template.FuncMap) (result []byte, err error) {
+	t, err := template.New(templName).Option("missingkey=error").Funcs(templateFuncMap).Parse(templ)
+	if err != nil {
+		return
+	}
+	out := bytes.NewBuffer([]byte{})
```
Comment (SchumacherFM) [inline pkg/build/build.go:331]:
```
var out bytes.Buffer
if err:=t.Exe...(&out); err != nil { ...} return out.Bytes(),nil 
```

File: pkg/providers/oplocal.go [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+func onePasswordCLIAuth(account string) (token string, err error) {
+	fmt.Printf("Enter the password for %s: ", account)
+	password, err := terminal.ReadPassword(0)
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("op", "signin", "--raw", "--account", account)
+
+	stdin, err := cmd.StdinPipe()
```
Comment (SchumacherFM) [inline pkg/providers/oplocal.go:120]: missing `stdin.Close`

**Review by SchumacherFM — CHANGES_REQUESTED** [review]

File: pkg/build/build.go (hunk 7 above)
Comment (smartinov) [reply pkg/build/build.go:188]: I'ts moved code, I promise i'll get back to refactor it at some point

**Pull request merged**